commands.go

```diff
@@ -1621,22 +1628,49 @@ func (c cmdable) SUnionStore(ctx context.Context, destination string, keys ...st
 //   - XAddArgs.Values = map[string]interface{}{"key1": "value1", "key2": "value2"}
 //
 // Note that map will not preserve the order of key-value pairs.
+// MaxLen/MaxLenApprox and MinID are in conflict, only one of them can be used.
 type XAddArgs struct {
-	Stream       string
-	MaxLen       int64 // MAXLEN N
+	Stream     string
+	NoMkStream bool
+	MaxLen     int64 // MAXLEN N
+
+	// Deprecated: use MaxLen+Approx, remove in v9.
 	MaxLenApprox int64 // MAXLEN ~ N
-	ID           string
-	Values       interface{}
+
+	MinID  string
+	Approx bool
+	Limit  int64
+	ID     string
+	Values interface{}
 }
```

Comment:
Perhaps group MaxLen, MinID, and Approx:

```
    MaxLen int64
    MinID string
    // Approx causes MaxLen and MinID to use "~" matcher (instead of "=").
    Approx bool
```